gomail_test.go

```diff
@@ -302,6 +302,55 @@ func TestAttachmentsOnly(t *testing.T) {
 	testMessage(t, msg, 1, want)
 }
 
+func TestBase64EncodedAttachment(t *testing.T) {
+	msg := NewMessage()
+	msg.SetHeader("From", "from@example.com")
+	msg.SetHeader("To", "to@example.com")
+	content1 := "hello"
+	content2 := "goodbye"
+	content3 := "something"
+	content1Buf := make([]byte, base64.StdEncoding.EncodedLen(len(content1)))
+	base64.StdEncoding.Encode(content1Buf, []byte(content1))
+	file1 := CreateFile("test.pdf", content1Buf)
+	file1.Encoding = Base64Encoded
+	file2 := CreateFile("test.zip", []byte(content2))
+	file2.Encoding = Base64Encoded
+	file3 := CreateFile("test.png", []byte(content3))
+	msg.Attach(file1)
+	msg.Attach(file2)
+	msg.Attach(file3)
+
+	want := message{
+		from: "from@example.com",
+		to:   []string{"to@example.com"},
+		content: "From: from@example.com\r\n" +
+			"To: to@example.com\r\n" +
+			"Content-Type: multipart/mixed; boundary=_BOUNDARY_1_\r\n" +
```

Comment:
Is there a need for a test that produces multipart/alternative, or somehow includes text/plain and text/html content?